### PR TITLE
Core: Add toggles_as_bools to options.as_dict

### PR DESCRIPTION
**Pull request by Exempt-Medic**
## What is this fixing or adding?

It's pretty common for a world to want their Toggle options to be Booleans and not Ints in slot_data, so this adds that as an option.

Might as well improve the typing while we're at it

## How was this tested?

Generations and printing slot_data using `options.as_dict` with Toggle and DefaultOnToggle options.
Apparently PyCharm also said no more spaces in empty lines 🤷

### CODE REVIEW

**Comment by NewSoupVi** [comment]
Seems nice to have to me

This is more of an "opinion" PR so like, probably need a couple more of those

File: Options.py [hunk 1]
```diff
@@ -1229,7 +1229,7 @@ class CommonOptions(metaclass=OptionsMetaProperty):
     progression_balancing: ProgressionBalancing
     accessibility: Accessibility
 
-    def as_dict(self, *option_names: str, casing: str = "snake") -> typing.Dict[str, typing.Any]:
+    def as_dict(self, *option_names: str, casing: str = "snake", toggles_as_bools: bool = False) -> typing.Dict[str, typing.Any]:
```
Comment (beauxq) [inline Options.py:1232]: I think this makes this line more than 120 characters.

Comment (beauxq) [reply Options.py:1232]:
```suggestion
    def as_dict(self,
                *option_names: str,
                casing: typing.Literal["snake", "camel", "pascal", "kebab"] = "snake",
                toggles_as_bools: bool = False) -> typing.Dict[str, typing.Any]:
```
This is how I would do it to make room for the improved `casing` typing that isn't relevant to this PR...

**Comment by alwaysintreble** [comment]
If we're going to this a kwarg to return current_key would be good too. I'm a bit against doing this because tools are more expensive to encode in slot_data than ints are, but the measurable difference is so minor and they're more convenient to use so eh. But ints are just as easy to parse on the client side.

**Comment by Exempt-Medic** [comment]
> If we're going to this a kwarg to return current_key would be good too. I'm a bit against doing this because bools are more expensive to encode in slot_data than ints are, but the measurable difference is so minor and they're more convenient to use so eh. But ints are just as easy to parse on the client side.

You could add that in another PR, I feel it's out of scope here, especially since `current_key` behavior varies by option type

File: Options.py [hunk 2]
```diff
@@ -1229,7 +1229,10 @@ class CommonOptions(metaclass=OptionsMetaProperty):
     progression_balancing: ProgressionBalancing
     accessibility: Accessibility
 
-    def as_dict(self, *option_names: str, casing: str = "snake") -> typing.Dict[str, typing.Any]:
+    def as_dict(self,
+                *option_names: str,
+                casing: typing.Literal["snake", "camel", "pascal", "kebab"] = "snake",
+                toggles_as_bools: bool = False) -> typing.Dict[str, typing.Any]:
         """
         Returns a dictionary of [str, Option.value]
```
Comment (beauxq) [inline Options.py:1238]: The other parameters are in the doc string. So now this one is missing from the doc string.

**Review by beauxq — APPROVED** [review]
I don't see problems with code.
I didn't test it.
I don't have an opinion on whether the feature should exist or not.

File: Options.py [hunk 3]
```diff
@@ -1256,6 +1260,8 @@ def as_dict(self, *option_names: str, casing: str = "snake") -> typing.Dict[str,
                 value = getattr(self, option_name).value
                 if isinstance(value, set):
                     value = sorted(value)
+                if toggles_as_bools and issubclass(type(self).type_hints[option_name], Toggle):
```
Comment (alwaysintreble) [inline Options.py:1263]: I think an elif is more appropriate here since a Toggle can't have a set as its value.

**Review by NewSoupVi — APPROVED** [review]

**Comment by NewSoupVi** [comment]
Oh wait, technically I was supposed to let Medic merge this themself now that they're a Core Maintainer... oh well :D

**Pull request merged**